lib/std/ascii.zig

```diff
@@ -281,3 +323,22 @@ test "indexOfIgnoreCase" {
 
     std.testing.expect(indexOfIgnoreCase("FOO foo", "fOo").? == 0);
 }
+
+/// Finds `substr` in `container`, case sensitive, starting at `start_index`.
+pub fn indexOfMatchCasePos(container: []const u8, start_index: usize, substr: []const u8) ?usize {
```

Comment:
These `MatchCase` functions should not exist: this one is just `mem.indexOfPos`

Reply:
You're right -- I'd wondered if there was a reason these did not exist already, and of course that is why.

This PR now just contains the addition of the allocUpperString and some docstring additions/changes. I admit it's a bit underwhelming. If it'd be helpful, I can go through and add docstrings to other functions that are currently missing them and turn this into a "documentation" PR.